src/cript/nodes/exceptions.py

```diff
@@ -4,16 +4,15 @@
 class CRIPTNodeSchemaError(CRIPTException):
     """
     Exception that is raised when a DB schema validation fails for a node.
-
-    This is a dummy implementation.
-    This needs to be way more sophisticated for good error reporting.
     """
 
-    def __init__(self):
-        pass
+    error_message: str
+
+    def __init__(self, error_message: str):
```

Comment:
I think this needs some love down the line.

Reply:
**heads up:** after using the db schema for a few days to edit a [giant JSON](https://criptscripts.org/cript_graph_json/cao_protein.html), I am not sure if we should show the users the [JSONSchema](https://json-schema.org/) errors because while they are nice for us to debug, the users might have a hard time understanding what went wrong. Since we hide the json from them and let them only work with the nodes, they might have a hard time understanding where this error came from.

The errors are not too bad though, they are workable, but not the most intuitive for a user

Example error
``` bash
jsonschema.exceptions.ValidationError: 'Whey protein isolate' is not of type 'object'

Failed validating 'type' in schema['properties']['collection']['items']['properties']['experiment']['items']['properties']['process']['items']['properties']['ingredient']['items']['properties']['material']:
    {'$id': 'MaterialPost',
     'additionalProperties': False,
     'properties': {'bigsmiles': {'type': 'string'},
                    'cas': {'type': 'string'},
                    'chem_formula': {'type': 'string'},
                    'component': {'items': {'$ref': 'MaterialPost'},
                                  'type': 'array'},
                    'computational_forcefield': {'$ref': 'ComputationalForcefieldPost'},
                    'created_at': {'$ref': 'Timestamp'},
                    'created_by': {'$ref': 'Creator'},
                    'inchi': {'type': 'string'},
                    'inchi_key': {'type': 'string'},
                    'keyword': {'$ref': 'MaterialKeyword'},
                    'locked': {'default': False, 'type': 'boolean'},
                    'model_version': {'$ref': 'ModelVersion'},
                    'mol_form': {'type': 'string'},
                    'name': {'maxLength': 2048,
                             'minLength': 2,
                             'type': 'string'},
                    'names': {'item': {'type': 'string'}, 'type': 'array'},
                    'node': {'items': {'enum': ['Material'],
                                       'type': 'string'},
                             'type': 'array'},
                    'notes': {'default': '', 'type': 'string'},
                    'parent_material': {'type': 'string'},
                    'property': {'items': {'$ref': 'PropertyPost'},
                                 'type': 'array'},
                    'pubchem_cid': {'type': 'integer'},
                    'public': {'default': False, 'type': 'boolean'},
                    'smiles': {'type': 'string'},
                    'uid': {'$ref': 'TempUID'},
                    'updated_at': {'$ref': 'Timestamp'},
                    'updated_by': {'$ref': 'Creator'},
                    'uuid': {'default': 'uuid()',
                             'readOnly': True,
                             'type': 'string'}},
     'required': ['node'],
     'type': 'object'}

On instance['collection'][0]['experiment'][0]['process'][0]['ingredient'][0]['material']:
    'Whey protein isolate'
```

This is for a JSON that was written like:
```json
"material": "Whey protein isolate"
````

instead of
```json
"material": {
  "node": ["Material"],
  "name": "Whey protein isolate",
  "uid": "_:Whey protein isolate"
}
```